### PR TITLE
Save returnUrl across requests

### PR DESCRIPTION
**Pull request by cmaish**
ReturnUrl and the hash should be persisted across a bunch of stuff on the account controller.  You should be able to create an account / log in / log in with a social provider and come back to the right location.

Updated images for the social login buttons to use fontawesome.
jabbr-login-buttons

### CODE REVIEW

File: JabbR/Views/Account/_social.cshtml [hunk 1]
```diff
@@ -17,11 +10,11 @@
                             <ul class="inline">
                                 @if (!Model.IsAlreadyLinked(provider))
                                 {
-                                    <li><a class="btn btn-provider" href="@Url.Content("~/authentication/redirect/" + provider.ToLower())"><i class="icon-@provider.ToLower()-logo"></i> @provider</a></li>
+                                    <li><a class="btn btn-provider" href="@Url.Content("~/authentication/redirect/" + provider.ToLower() + Html.RequestQuery())"><i class="icon-@provider.ToLower()"></i> @provider</a></li>
                                 }
                                 else
                                 {
-                                    <li><div class="provider"><i class="icon-@provider.ToLower()-logo"></i> @provider</div><button type="submit" class="btn-unlink" name="provider" value="@provider.ToLower()" title="@LanguageResources.Account_UnlinkIdentityProvider"><i class="icon-unlink"></i></button></li>
```
Comment (davidfowl) [inline JabbR/Views/Account/_social.cshtml:24]: ?

File: JabbR/Views/Account/_layout.cshtml [hunk 2]
```diff
@@ -27,7 +34,28 @@
     <script src="@Url.Content("~/Scripts/jquery-2.0.3.min.js")"></script>
     <script type="text/javascript">
         (function($) {
-            $('input.error').closest('.control-group').addClass('error');
```
Comment (davidfowl) [inline JabbR/Views/Account/_layout.cshtml:30]: Use 4 spaces not 2

File: JabbR/Views/Account/_social.cshtml (hunk 1 above)
Comment (cmaish) [inline JabbR/Views/Account/_social.cshtml:24]: This is related to the classes that were pulled out of the css.  Previously we'd use an image for the file, but there's an equivalent logo available in fontawesome which we use now.

**Comment by hnrkndrssn** [comment]
Can we have the icons centred together with the text (like it currently is on the login page on jabbr.net)? It looks a bit odd to me having the icon left justified when the text is centred.

**Comment by cmaish** [comment]
jabbr-login-buttons-new

Made social buttons look like the ones that are in the current version of jabbr again.  New buttons at top, existing jabbr.net buttons at bottom.

**Comment by hnrkndrssn** [comment]
Looks good to me :+1:

**Comment by cmaish** [comment]
should fix #737 and #652

**Pull request merged**